tests/suites/test_suite_ssl.function

```diff
@@ -3680,6 +3737,42 @@ void handshake_serialization( )
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C:!MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_PKCS1_V15:MBEDTLS_RSA_C:MBEDTLS_ECP_DP_SECP384R1_ENABLED */
+void handshake_fragmentation( int mfl, int expected_srv_hs_fragmentation, int expected_cli_hs_fragmentation)
+{
+    handshake_test_options options;
+    log_pattern srv_pattern, cli_pattern;
+    size_t srv_hs_fragmented = 0;
+    size_t cli_hs_fragmented = 0;
+
+    srv_pattern.pattern = cli_pattern.pattern = "found fragmented DTLS handshake";
+    srv_pattern.counter = &srv_hs_fragmented;
+    cli_pattern.counter = &cli_hs_fragmented;
+
+    init_handshake_options( &options );
+    options.dtls = 1;
+    options.mfl = mfl;
+    options.srv_auth_mode = MBEDTLS_SSL_VERIFY_REQUIRED;
+    options.srv_log_obj = &srv_pattern;
+    options.cli_log_obj = &cli_pattern;
+    options.srv_log_fun = log_analyzer;
+    options.cli_log_fun = log_analyzer;
+
+    perform_handshake( &options );
+
+    /* Test if the server received fragmented handshake */
+    if( expected_srv_hs_fragmentation )
+    {
+        TEST_ASSERT( srv_hs_fragmented >= 1 );
```

Comment:
Is this sufficient? Why won't we check for exact values, passed as a test parameter?

Reply:
I was thinking about it too, but it is pretty hard to determinate how much data should be transferred during handshake and in how many fragments. I decided that the fact of any fragmentation is sufficient enough for this test.

Reply:
What do you mean by `hard to determine`? Is it time consuming for the programmer, is it non-deterministic, or is there no way to tell the exact number of fragments? I can see that in `ssl-opt.sh` application data tests check the exact number and size of fragments, while DTLS handshake fragmentation tests only check if the handshake message was fragmented, so I'm inclined towards accepting it, but I would also like to know the background a bit more.

Reply:
If we want to send application data, we decide how much of them should be. For handshake the message length is unknown it should be estimate. It is related with the certificates, allowed cipher suites and etc. I think it is not easy to estimate the biggest message size during handshake. I could hardcoded the current number of fragments with which this test is performed (if it is good enough)